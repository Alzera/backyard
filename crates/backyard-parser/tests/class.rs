use backyard_parser::parse_eval;

#[test]
fn basic() {
  let asts = parse_eval("class A {\n}").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn modifiers() {
  let asts = parse_eval("readonly final class A {\n}").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn extended() {
  let asts = parse_eval("class A extends B implements C, D {\n}").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn anonymous() {
  let asts = parse_eval("$a = new class {\n};").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn parameter() {
  let asts = parse_eval("$a = new class($b) {\n};").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn anonymous_extended() {
  let asts = parse_eval("$a = new class($b) extends B implements C, D {\n}").unwrap();
  insta::assert_yaml_snapshot!(asts);
}
