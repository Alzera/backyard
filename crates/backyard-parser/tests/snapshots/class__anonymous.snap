---
source: crates/backyard-parser/tests/class.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: assignment
    left:
      node_type: variable
      name:
        node_type: identifier
        name:
          - 97
        loc:
          start:
            line: 1
            column: 0
            offset: 0
          end:
            line: 1
            column: 1
            offset: 1
        leadings: ~
        trailings: ~
      loc:
        start:
          line: 1
          column: 0
          offset: 0
        end:
          line: 1
          column: 1
          offset: 1
      leadings: ~
      trailings: ~
    operator: Default
    right:
      node_type: new
      statement:
        node_type: anonymous_class
        parameters: []
        extends: ~
        implements: []
        body:
          node_type: block
          statements: []
          loc:
            start:
              line: 1
              column: 15
              offset: 15
            end:
              line: 2
              column: 0
              offset: 17
          leadings: ~
          trailings: ~
        loc:
          start:
            line: 1
            column: 9
            offset: 9
          end:
            line: 2
            column: 0
            offset: 17
        leadings: ~
        trailings: ~
      loc:
        start:
          line: 1
          column: 5
          offset: 5
        end:
          line: 2
          column: 0
          offset: 17
      leadings: ~
      trailings: ~
    loc:
      start:
        line: 1
        column: 3
        offset: 3
      end:
        line: 2
        column: 0
        offset: 17
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 2
    column: 1
    offset: 18
leadings: ~
trailings: ~
